# Laboratorio 14: DISTRIBUCIÓN BIVARIADA. DISTRIBUCIÓN MARGINAL. DISTRIBUCIÓN CONDICIONAL

## Integrante:

- Aranda Huerta, Milene Natalia
- Escriba Flores, Daniel Agustin

--- 

In [1]:
# Importamos las librerias necesarias

import pandas as pd 
import numpy as np 
import math 

--- 
## Pregunta 1: 

Se selecciona al azar dos libros de texto de un anaquel que contiene tres libros de estadística, dos de matemáticas y tres de física. Si 𝑋 es el número de libros de estadística e 𝑌 es el número de libros de matemáticas realmente escogidos. En un notebook de Jupyter y utilizando las librerías ‘pandas’ y ‘numpy’, y la librería ‘math’, haga lo siguiente:

### a. Halle la función de distribución conjunta de 𝑋 e 𝑌

In [2]:
x = np.array([0,0,1,1,2,0])
print("X=",x)
y = np.array([0,1,0,1,0,2])
print("Y=",y)
z = np.array([2,1,1,0,0,0])
print("Z=",z) 

X= [0 0 1 1 2 0]
Y= [0 1 0 1 0 2]
Z= [2 1 1 0 0 0]


In [3]:
# Definimos los valores

E, M, F = 3, 2, 3
T = E+M+F
n = math.comb(T,2) # Combinaciones de T en 2
print("Total maneras =",n)
p = len(x)


Total maneras = 28


In [4]:
# Hallamos las maneras de cada situacion

Maneras = np.zeros(p)
for i in range(p):
    Maneras[i] = math.comb(E,x[i]) * math.comb(M,y[i]) * math.comb(F,z[i])
print(Maneras)

[3. 6. 9. 6. 3. 1.]


In [5]:
fxy = Maneras/n

print("f(x,y)=",fxy)


f(x,y)= [0.10714286 0.21428571 0.32142857 0.21428571 0.10714286 0.03571429]


In [6]:
# Funcion de distribucion conjunta
tabla = pd.DataFrame({"X":x,"Y":y,"f(x,y)":fxy})
tabla

X  Y    f(x,y)
0  0  0  0.107143
1  0  1  0.214286
2  1  0  0.321429
3  1  1  0.214286
4  2  0  0.107143
5  0  2  0.035714

###  B. Halle la función de distribución marginal de 𝑋 = 1

In [7]:
# Funcion de distribucion marginal x=1
s=0
for i in range(p):
    if x[i]==1:
        s=s+fxy[i]
fx1 = round(s,4)
print(fx1)

0.5357


$f_1(1)=0.5357$

###  C. Halle la función de distribución condicional de 𝑌 dado 𝑋 = 1.

In [8]:
# Funcion de distribucion condicional de y dado x=1
fy_x = (np.delete(fxy,(0,1,4,5)))/fx1
fy_x = np.append(fy_x,0)
print(fy_x)

[0.600016   0.40001067 0.        ]


$f(1|0)=0.60016$  
$f(1|1)=0.40001$  
$f(1|2)=0$   

---
## Pregunta 2:

Suponga que 𝑋, el precio de cierta mercadería (en soles) e 𝑌, sus ventas totales (en 10 000 unidades), son variables aleatorias, cuya función de distribución conjunta es la mostrada

$f(x, y) =
\begin{cases} 
    5x e^{-0.3y}, & 0.20 < x < 0.40, \, y > 0 \\ 
    0, & \text{c.c.}
\end{cases}$

In [9]:
# Librerias para este caso

from sympy import integrate
from sympy.abc import x, y
from sympy import exp


### a. Calcule la probabilidad de que el precio será menos de 30 céntimos y las ventas excedan 20 000 unidades.

In [10]:
fxy =5*x*exp(-0.3*y)
fxy

5*x*exp(-0.3*y)

Debemos calcular , pero sabemos que .𝑃(𝑥 < 0. 30, 𝑦 > 2) 0. 20 < 𝑥 < 0. 40, 𝑦 > 0 Entonces: 𝑃(0. 20 < 𝑥 < 0. 30, 𝑦 > 2)

In [11]:
# 𝑃(0. 20 < 𝑥 < 0. 30, 𝑦 > 2)

I = integrate(fxy,(y,2,math.inf))
I

9.14686060156711*x

In [12]:
P = integrate(I,(x,0.2,0.3))
P

0.228671515039178

$𝑃(0. 20 < 𝑥 < 0. 30, 𝑦 > 2) = 0.22867$

### b. Halle la función de densidad marginal de 𝑋 e 𝑌

In [13]:
# Función de densidad marginal de X
fx = integrate(fxy,(y,0,math.inf))
fx

16.6666666666667*x

In [14]:
# Función de densidad marginal de Y
fy = integrate(fxy,(x,0.2,0.4))
fy

0.3*exp(-0.3*y)

$f_1(x)=16.6667x,\hspace{1cm}0.2<x<0.4$    
$f_2(Y)=0.3e^{-0.3y},\hspace{1cm}y > 0$


### c. Halle la función de densidad condicional de 𝑋 dado 𝑌 = 𝑦 y la función de densidad condicional de 𝑌 dado 𝑋 = 𝑥

In [15]:
# Función de densidad condicional de X dado Y = y
fx_y = fxy/fy
fx_y

16.6666666666667*x

In [16]:
# Función de densidad condicional de Y dado X = x
fy_x = fxy/fx
fy_x

0.3*exp(-0.3*y)

$f_1(X|Y)=16.6667x,\hspace{1cm}0.2<x<0.4$    
$f_2(Y|X)=0.3e^{-0.3y},\hspace{1cm}y > 0$
